chart/templates/controller.yaml

```diff
@@ -52,52 +52,59 @@ rules:
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRoleBinding
 metadata:
-  name: {{.Values.wds_cp_name}}-transport-controller-rolebinding
+  name: {{.Values.wds_cp_name}}-transport-controller
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: ClusterRole
-  name: {{.Values.wds_cp_name}}-transport-controller-role
+  name: {{.Values.wds_cp_name}}-transport-controller
 subjects:
   - kind: ServiceAccount
-    name: transport-controller-sa
-    namespace: '{{.Values.wds_cp_name}}-system'
+    name: transport-controller
+    namespace: {{.Release.Namespace}}
 ---
 apiVersion: v1
 kind: ConfigMap
 metadata:
   name: transport-controller-config
-  namespace: {{.Values.wds_cp_name}}-system
+  namespace: {{.Release.Namespace}}
 data:
-  get-kubeconfig.sh: |
-    #!/bin/bash
-    # this script receives a ControlPlane name and a parameter
-    # that determines whether to extract the ControlPlane's in-cluster kubeconfig
-    # or the external kubeconfig (if set to "true", the first will be retrieved).
-    # The function returns the requested kubeconfig's content in base64.
-    # it assumes the kubeconfig context is set to access the hosting cluster.
-
-    cpname="$1"
-    get_incluster_key="$2"
-
-    key=""
-    if [[ "$get_incluster_key" == "true" ]]; then
-      key=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.inClusterKey}');
-    else
-      key=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.key}');
+  get-config.sh: |
+    #!/bin/env bash
+    cp_name="${1%"-system"}" # cp name or cp namespace
+    guess_its_name="$2"
+    if [ "$cp_name" == "" ] ; then
+      if [ "$guess_its_name" == "true" ] ; then
+        for cp in `kubectl get controlplane -o name`; do
+            cp=${cp##*/}
+            if kubectl get controlplane $cp -o=jsonpath='{.metadata.labels}' | grep "imbs" &> /dev/null ; then

```

Comment:
Grepping through all the labels for "imbs" is a bit fragile, and it will be worse when we change that to "its". I suggest using the following instead.

```
mspreitz@mjs13 ocm-transport-plugin % kubectl --context kind-kubeflex get controlplanes imbs1 -o jsonpath="{.metadata.labels['kflex\.kubestellar\.io/cptype']}"
imbs
```

Reply:
@MikeSpreitzer thank you, I tried several combinations but could not get it to work so I surrendered to grep.
I will change it in the new version to use your filter.

Reply:
Yes, I also spent a lot of time being stymied by this undocumented mystery. One day I stumbled across someone else's code using the solution.